.changelog/29181.txt

```diff
@@ -0,0 +1,23 @@
+```release-note:enhancement
+resource/aws_ec2_fleet: Add computed `fleet_instance_set`, `fleet_state`, `fulfilled_capacity`, and `fulfilled_on_demand_capacity` arguments
+```
+
+```release-note:enhancement
+resource/aws_ec2_fleet: Add `launch_template_config.override.instance_requirements.allowed_instance_types`,  launch_template_config.override.instance_requirements.network_bandwidth_gbps` arguments
+```
+
+```release-note:enhancement
+resource/aws_ec2_fleet: Add `on_demand_options.capacity_reservation_options`,`on_demand_options.max_total_price`, `on_demand_options.min_target_capacity`, `on_demand_options.single_availability_zone`, `on_demand_options.single_instance_type` arguments
+```
+
+```release-note:enhancement
+resource/aws_ec2_launch_template: Add expand and flatten support for `instance_requirements.allowed_instance_types`,  `instance_requirements.network_bandwidth_gbps_request`
```

Comment:
The `aws_ec2_fleet` resource referenced the `flattenInstanceRequirements` function in `aws_ec2_launch_template`, prompting this update. Acceptance testing for all resources that call this function is included in the PR.